staging/src/k8s.io/client-go/openapi/openapitest/fakeclient.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapitest
+
+import (
+	"k8s.io/client-go/openapi"
+)
+
+// FakeClient implements openapi.Client interface, with hard-coded
+// return values, including the possibility to force errors.
+type FakeClient struct {
+	// Hard-coded paths to return from Paths() function.
+	PathsMap map[string]openapi.GroupVersion
+	// Hard-coded returned error.
+	ForcedErr error
+}
+
+// Validate FakeClient implements openapi.Client interface.
+var _ openapi.Client = &FakeClient{}
+
+// NewFakeClient returns a fake openapi client with an empty PathsMap.
+func NewFakeClient() *FakeClient {
+	return &FakeClient{PathsMap: make(map[string]openapi.GroupVersion)}
+}
+
+// Paths returns stored PathsMap field, creating an empty one if
+// it does not already exist. If ForcedErr is set, this function
+// returns the error instead.
+func (f FakeClient) Paths() (map[string]openapi.GroupVersion, error) {
+	if f.ForcedErr != nil {
+		return nil, f.ForcedErr
+	}
+	return f.PathsMap, nil
+}
+
+// FakeGroupVersion implements openapi.GroupVersion with hard-coded
+// return GroupVersion specification bytes. If ForcedErr is set, then
+// "Schema()" function returns the error instead of the GVSpec.
+type FakeGroupVersion struct {
+	// Hard-coded GroupVersion specification
+	GVSpec []byte
```

Comment:
Could we have a way to map the data to the content type?

Reply:
My understanding is that we will not support protobuf--we will only support JSON. With that the case, I don't think we need to map the content type.